collect_app/src/main/java/org/odk/collect/android/preferences/screens/ProjectDisplayPreferencesFragment.kt

```diff
@@ -125,11 +130,16 @@ class ProjectDisplayPreferencesFragment :
     override fun onPreferenceChange(preference: Preference, newValue: Any): Boolean {
         val (uuid, name, icon, color) = currentProjectProvider.getCurrentProject()
         when (preference.key) {
-            PROJECT_NAME_KEY -> projectsRepository.save(
-                Project.Saved(
-                    uuid, newValue.toString(), icon, color
+            PROJECT_NAME_KEY -> {
```

Comment:
Could all this be done in `SettingsChangeHandler` instead so that the work only has to be done in one place for all settings paths?

Reply:
No because we don't store project details in settings so `SettingsChangeHandler` is not called as well.

Reply:
Ahhh, I see. That's really too bad. This feels dangerous because  it's the kind of thing where we could add additional paths. But let's go with it for now. At least it's not critical behavior.

Reply:
Even if we stored those details in settings it would be a problem because in `SettingsChangeHandler` we have just access to the new value but I also need the old one to delete the existed file.